src/utilities/focus.ts

```diff
@@ -158,6 +159,61 @@ export function focusLastKeyboardFocusableNode(
   return false;
 }
 
+function getMenuFocusableDescendants(element: HTMLElement) {
+  return element.querySelectorAll(
+    MENUITEM_FOCUSABLE_SELECTORS,
+  ) as NodeListOf<HTMLElement>;
+}
+
+function getCurrentFocusedElementIndex(
+  allFocusableChildren: NodeListOf<HTMLElement>,
+  currentFocusedElement: HTMLElement,
+) {
+  let currentItemIdx = 0;
+
+  for (const focusableChild of allFocusableChildren) {
+    if (focusableChild === currentFocusedElement) {
+      break;
+    }
+    currentItemIdx++;
+  }
+  return currentItemIdx;
+}
+
+export function wrapFocusPreviousFocusableMenuItem(
+  parentElement: HTMLElement,
+  currentFocusedElement: HTMLElement,
+) {
+  const allFocusableChildren = getMenuFocusableDescendants(parentElement);
+  const currentItemIdx = getCurrentFocusedElementIndex(
+    allFocusableChildren,
+    currentFocusedElement,
+  );
+
+  if (currentItemIdx === 0) {
+    allFocusableChildren[allFocusableChildren.length - 1].focus();
+  } else {
+    allFocusableChildren[currentItemIdx - 1].focus();
+  }
+}
+
+export function wrapFocusNextFocusableMenuItem(
+  parentElement: HTMLElement,
+  currentFocusedElement: HTMLElement,
+) {
+  const allFocusableChildren = getMenuFocusableDescendants(parentElement);
+  const currentItemIdx = getCurrentFocusedElementIndex(
+    allFocusableChildren,
+    currentFocusedElement,
+  );
+
+  if (currentItemIdx === allFocusableChildren.length - 1) {
```

Comment:
Your solution works great! Just because we talked about it, this would be the modulus version of the if/else:
` allFocusableChildren[
    (currentItemIdx + 1) % allFocusableChildren.length
  ].focus();`